File: packetLoss.m
clear
clc
qualities = [0,  0, 75, 0, 0];
oneHotQuality = [ones(1,qualities(5)), ones(1,qualities(1)), ones(1,qualities(2)), ones(1,qualities(3)), ones(1,qualities(4))];
sizes = [834136, 211384, 54760, 14464, 3616];
max_data = sum(qualities.*sizes)
packetSize = 1000;
numOfPackets = fix(max_data/packetSize);
bandSizes = fix(sizes/packetSize);

for i = 1:12
    filename = "UP_loss"+int2str(i)+".txt"
    fileID = fopen('robustness\loss\'+filename, 'a');
    numOfLoss = fix(i*numOfPackets/100);
    theZeros = zeros(1, numOfLoss);
    theOnes = ones(1, (numOfPackets - numOfLoss));
    indices = [theZeros theOnes];
    for j = 1:10
            indices = indices(randperm(length(indices)));
            ind = 0;
            while(ind<length(indices))
                ind = ind + 1;
                if(indices(ind) == 0)
                    [quality, number] = findzero(ind, qualities, bandSizes);
                    fprintf(fileID, '%d %d\n', quality, number);
                end
            end
    end
    fclose(fileID)
end

function [quality, number] = findzero(ind, qualities, bandSizes)
    offset = 0;
    for qual = 1:5
        for i = 1:qualities(qual)
            if(ind <= i*bandSizes(qual) + offset)
                quality = qual;
                number = i;
                return
            end
        end
        offset = offset + qualities(qual)*bandSizes(qual);
    end
    quality = 5;
    number = qualities(5);            
end

File: packetLoss.py
def main():
    # [ 8  3 15 62 14]
    # (834136, 211384, 54760, 14464, 3616)
    qualities = [8,  3, 15, 62, 14]
    oneHotQuality = np.concatenate((np.ones(qualities[0]), np.ones(qualities[1]), np.ones(qualities[2]), np.ones(qualities[3]), np.ones(qualities[4])))
    sizes = [834136, 211384, 54760, 14464, 3616]
    max_data = np.sum(np.multiply(qualities, sizes))
    packetSize = 1000
    numOfPackets = max_data // packetSize
    for i in range(12):
        numOfLoss = i*numOfPackets // 100
        theZeros = np.zeros(numOfLoss)
        theOnes = np.ones(numOfPackets - numOfLoss)
        indices = np.concatenate((theZeros, theOnes))
        for j in range(30):
            indices = np.random.shuffle(indices)
            print(indices)
            for k in range(indices):
                if(indices[k] == 0):
                    if(k <= qualities[0]*sizes[0])
                        for k1 in range(qualities[0]):
                            if(k<=k1*sizes[0]):
                                oneHotQuality[k1] = 0
                
                    
                    
            
    packetLossindex = np.random.randint(2, size = numOfPackets)
